packages/matchbox/src/components/Toggle/Toggle.js

```diff
@@ -1,68 +1,75 @@
-import React, { Component } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
-import classnames from 'classnames';
+import { visuallyHidden } from '../../styles/helpers';
+import styled from 'styled-components';
+import { deprecate } from '../../helpers/propTypes';
+import { ScreenReaderOnly } from '../ScreenReaderOnly';
+import { margin } from 'styled-system';
+import { createPropTypes } from '@styled-system/prop-types';
+import { toggle, input, StyledIndicator, StyledOutline } from './styles';
+import { pick } from '@styled-system/props';
+import { omit } from '../../helpers/systemProps';
 
-import styles from './Toggle.module.scss';
+const StyledToggle = styled('label')`
+  ${toggle}
+  ${margin}
+`;
 
-class Toggle extends Component {
-  static displayName = 'Toggle';
+const StyledInput = styled('input')`
+  ${visuallyHidden}
+  ${input}
+`;
 
-  static propTypes = {
-    id: PropTypes.string.isRequired,
-    checked: PropTypes.oneOfType([
-      PropTypes.bool,
-      PropTypes.string
-    ]),
-    compact: PropTypes.bool,
-    value: PropTypes.oneOfType([
-      PropTypes.bool,
-      PropTypes.string
-    ]),
-    disabled: PropTypes.bool,
-    onChange: PropTypes.func,
-    onFocus: PropTypes.func,
-    onBlur: PropTypes.func
-  };
+function Toggle(props) {
+  const {
+    id,
+    value,
+    checked,
+    disabled,
+    onChange,
+    onFocus,
+    onBlur,
+    label,
+    required,
+    ...rest
+  } = props;
+  const systemProps = pick(rest);
+  const componentProps = omit(rest, [margin.propNames]);
 
-  render() {
-    const {
-      id,
-      value,
-      checked,
-      compact,
-      disabled,
-      onChange,
-      onFocus,
-      onBlur,
-      ...rest
-    } = this.props;
-
-    const labelMarkup = compact ? null : (
-      <div className={styles.Labels}>
-        <span>On</span>
-        <span>Off</span>
-      </div>
-    );
-
-    return (
-      <label htmlFor={id} className={classnames(styles.Toggle, compact && styles.compact, disabled && styles.disabled)}>
-        <input
-          id={id}
-          value={value}
-          checked={checked}
-          disabled={disabled}
-          className={styles.Input}
-          onChange={onChange}
-          onFocus={onFocus}
-          onBlur={onBlur}
-          type='checkbox'
-          {...rest} />
-        <span className={styles.Outline}></span>
-        {labelMarkup}
-        <span className={styles.Indicator}></span>
-      </label>
-    );
-  }
+  return (
+    <StyledToggle htmlFor={id} disabled={disabled} {...systemProps}>
+      {label && <ScreenReaderOnly>{label}</ScreenReaderOnly>}
```

Comment:
Hmm, this won't always work, as sometimes the label is actually visible - _but_ we can save that for when component APIs update

Reply:
This would be part of `ToggleBlock` implementation though right? I think I intended this prop to only be used when another label isn't, eg. toggles inside table rows.

Reply:
That makes sense to me for sure.